### DISCUSSION TITLE
Ability to alter water level

### DISCUSSION THREAD

**Discussion by hokfujow** [vinteo/hass-opensprinkler#219] [Ideas]
Hi - I wanted to thank the contributors to this project for this great integration of HA with Opensprinkler.  For the most part, I'm happy with the native interface and programming of OS via web and mobile app.  However, I've found the weather adjustments to be less optimal as the algorithm doesn't take the forecast into account and also seems to only consider rain for the previous 24 hours.  So if we've had 3 days of pouring rain and then one day of sun, OS will not reduce the watering level.

I have a rain sensor which will probably help with the previous precipitation and of course, I can always add a manual rain delay, but I'd like to be able to have more sophisticated weather data usage.

Are there any plans for the integration to be able to set the Watering level via API?

Thanks!

**Comment by Nick-Adams-AU** [comment]
Unfortunately I don't think OS has the ability to set the watering level via API. This question might be best directed at the OS GitHub.

**Reply by hokfujow** [reply]
Oh actually I just realized there is already a HA service defined to set watering level! There are instructions for using the service in the readme file.  What's the protocol here ?  Should I mark this as closed or resolved?

**Comment by vinteo** [comment]
Yea there is already a service for it
